crates/ruma-common/src/events/room/member/change.rs

```diff
@@ -144,6 +146,7 @@ pub(super) fn membership_change<'a>(
         (St::Knock, St::Invite) => Ch::KnockAccepted,
         (St::Knock, St::Leave) if sender == state_key => Ch::KnockRetracted,
         (St::Knock, St::Leave) => Ch::KnockDenied,
+        (a, b) if a == b => Ch::None,
```

Comment:
One thing that was `NotImplemented` before but `None` now is two equal unknown membership states. I gut feeling is that that's fine, but maybe it isn't?

Reply:
Should be fine, there's effectively no state change